### PR TITLE
integration test for collection upload

### PR DESCRIPTION
**Pull request by thailekha**
Motivation:
Integration test for https://issues.jboss.org/browse/RHMAP-11591

JIRA: https://issues.jboss.org/browse/RHMAP-13466

### CODE REVIEW

File: src/endpoints/http/collections/index.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ export function collectionsHandler(router) {
         res.status(CREATED).end();
       })
       .catch(err => {
+
```
Comment (sedroche) [inline src/endpoints/http/collections/index.js]: can you remove this whitespace here and the other places?

File: test/integration/test_collections.js [hunk 2]
```diff
@@ -66,6 +69,33 @@ module.exports = {
            expect(res.text).to.equal('test1,test2 collection(s) deleted');
            done();
          });
+    },
+
+    'test_collection_upload': function(done) {
```
Comment (sedroche) [inline test/integration/test_collections.js]: I like the test but you can remove the `ext` parameter by doing:
```javascript
    'test_collection_upload': function(done) {
      const test = exts => {
        if (!exts.length) {
          return done();
        }

        const ext = exts.shift();
        chai.request(SERVER_URL)
          .post(`${PATH_PREFIX}/collections/upload`)
          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
            expect(res).to.have.status(201);

            mongodbClient.dropCollection('import', test.bind(null, exts));
          });
      };

      test(['json','csv','bson']);
    }
```

Also can you fix the indentation on the function chain? 2 spaces like in my example above.

File: test/integration/test_collections.js [hunk 3]
```diff
@@ -66,6 +69,33 @@ module.exports = {
...
+          mongodbClient.dropCollection('import', err => {
+            expect(err).to.be.null;
+
+            test(exts.shift(), exts);
+          });
+        });
+      };
+
+      const extensions = ['json','csv','bson'];
+      test(extensions.shift(), extensions);
```
Comment (sedroche) [inline test/integration/test_collections.js]: I just want to make sure we are ok with using the recursive method here. It's slightly different to how other tests are written.

The other option is using async and would look something like
```javascript
    'test_collection_upload': function(done) {
      const test = (ext, cb) => {
        chai.request(SERVER_URL)
          .post(`${PATH_PREFIX}/collections/upload`)
          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
            expect(res).to.have.status(201);
            mongodbClient.dropCollection('import', cb);
          });
      };

      async.eachSeries(['json','csv','bson'],  test, done);
    }
```

Thoughts @pb82 @rachael-oregan ?

**Review by sedroche — CHANGES_REQUESTED** [review]
Couple of comments

File: api-docs/swagger.yaml [hunk 4]
```diff
@@ -92,6 +92,43 @@ paths:
...
             $ref: '#/definitions/ErrorResponse'
+  '/{domain}/{envId}/{appGuid}/data/collections/upload':
+    parameters:
+      - $ref: "#/parameters/domain"
+      - $ref: "#/parameters/envId"
+      - $ref: "#/parameters/appGuid"
+    post:
+      summary: Upload collection(s)
+      description: |
+        Upload json, bson, or csv file to import collection(s)
```
Comment (sedroche) [inline api-docs/swagger.yaml]: Change this to "Upload json, bson, or csv file collection(s)"

File: api-docs/swagger.yaml [hunk 5]
```diff
@@ -92,6 +92,43 @@ paths:
...
+      operationId: uploadCollection
```
Comment (sedroche) [inline api-docs/swagger.yaml]: I misunderstood this when you asked. Can you change this to `insertCollection` please?

File: test/integration/test_collections.js [hunk 6]
```diff
@@ -62,10 +64,32 @@ module.exports = {
...
+          .set('Authorization', `Bearer ${TOKEN}`)
+          .end((err, res) => {
+            expect(err).to.be.null;
+            expect(res).to.have.status(statusCodes.CREATED);
+
+            mongodbClient.dropCollection('import', test.bind(null, exts));
+          });
+      };
+
+      test(['json','csv','bson']);
```
Comment (sedroche) [inline test/integration/test_collections.js]: I just want to make sure we are ok with using the recursive method here. It's slightly different to how other tests are written and we might want to maintain consistent code.

The other option is using async and would look something like
```javascript
    'test_collection_upload': function(done) {
      const test = (ext, cb) => {
        chai.request(SERVER_URL)
          .post(`${PATH_PREFIX}/collections/upload`)
          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
            expect(res).to.have.status(201);
            mongodbClient.dropCollection('import', cb);
          });
      };

      async.eachSeries(['json','csv','bson'],  test, done);
    }
```

Thoughts @pb82 @rachael-oregan ?

Comment (rachael-oregan) [reply test/integration/test_collections.js]: I think we should try keep the code consistent, it's more readable

Comment (pb82) [reply test/integration/test_collections.js]: I also think that using `async` is a bit nicer. But not a big deal.

**Review by pb82 — APPROVED** [review]
LGTM, just a comment

**Pull request merged**